miio/gateway/gatewaydevice.py

```diff
@@ -32,4 +32,5 @@ def __init__(
             )
 
         self._gateway = parent
-        super().__init__(ip, token, start_id, debug, lazy_discover)
```

Comment:
See my last edit in my comment. However, I think it may make sense to cleanup this class to not to inherit from `Device` at all (as they are not independent devices) but rather create necessary wrappers to perform I/O through the `parent` reference?

Reply:
yea I agree, will convert it now